### ISSUE TITLE
Share database connections in celery workers

### ISSUE THREAD

**Issue by hardbyte** [data61/anonlink-entity-service#289]
It takes some work to connect to the postgresql database and almost every worker task does that multiple times - usually just relying on Python's garbage collection for closing connections. Instead we might manage that state in a per worker pool, or since celery itself maintains a pool of worker processes, we could just init a connection along with each worker. 

```
db_conn = None

@worker_process_init.connect
def init_worker(**kwargs):
    global db_conn
    logger.info('Initializing database connection for worker.')
    db_conn = connect_db()


@worker_process_shutdown.connect
def shutdown_worker(**kwargs):
    global db_conn
    if db_conn:
        logger.info('Closing database connection for worker.')
        db_conn.close()

```

- https://stackoverflow.com/questions/14526249/celery-worker-database-connection-pooling
- https://code.i-harness.com/en/q/2263d77

**Comment by hardbyte** [comment]
Possible bug sighting in the wild during a 1M x 10M match:


```
 2019-06-05T05:58:05.971235Z [debug    ] Connecting to redis            [entityservice.cache] 
2019-06-05T05:58:05.971634Z [debug    ] Looking up read only redis slave using sentinel protocol [entityservice.cache] 
2019-06-05T05:58:05.985530Z [warning  ] fallback connecting to default postgres db [entityservice.database.util] database=postgres host=june-testing-anonlink-db user=postgres
2019-06-05T05:58:06.012383Z [warning  ] Can't connect to database      [entityservice.database.util] 
[2019-06-05 05:58:06 +0000] [9] [ERROR] Socket error processing request.
Traceback (most recent call last):
  File "/var/www/entityservice/database/util.py", line 38, in connect_db
    conn = psycopg2.connect(database=db, user=user, password=pw, host=host)
  File "/usr/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  sorry, too many clients already
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/var/www/entityservice/database/util.py", line 41, in connect_db
    conn = psycopg2.connect(database='postgres', user=user, password=pw, host=host)
  File "/usr/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: FATAL:  sorry, too many clients already 
```

**Comment by gusmith** [comment]
The observed bug observation had its own issue #385 .
The issue will be closed by #405

**Comment by gusmith** [comment]
Closed by #405
